key.core/src/main/java/de/uka/ilkd/key/logic/VariableNamer.java

```diff
@@ -48,6 +48,8 @@
 public abstract class VariableNamer implements InstantiationProposer {
     private static final Logger LOGGER = LoggerFactory.getLogger(VariableNamer.class);
 
+    public static final char TEMPORARY_VAR_SEPARATOR = '#';
```

Comment:
I see the point that it is necessary to move this constant, but I think the name is misleading. The original one was `TempIndProgramElementName.SEPARATOR`, which indicates that it is used to mark a temporary **index**.
Maybe we should call it "TEMP_INDEX_SEPARATOR" and also provide a JavaDoc here.

Reply:
I have renamed the constant and added some JavaDoc.